File: YoutubeQueuer/YoutubeQueuer.Lib/Services/YoutubeVideosService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Google.Apis.Auth.OAuth2;
using Google.Apis.YouTube.v3;
using YoutubeQueuer.Lib.Models;
using YoutubeQueuer.Lib.Providers.Abstract;
using YoutubeQueuer.Lib.Services.Abstract;

namespace YoutubeQueuer.Lib.Services
{
    internal class YoutubeVideosService : IYoutubeVideosService
    {
        private readonly IYoutubeServiceProvider _serviceProvider;

        public YoutubeVideosService(IYoutubeServiceProvider serviceProvider)
        {
            _serviceProvider = serviceProvider;
        }

        public IEnumerable<YoutubeVideoModel> GetLatestVideosFromChannel(string channelId, DateTime startDate, UserCredential credential)
        {
            const string parts = "id,snippet";
            var service = _serviceProvider.GetYoutubeService(credential);

            var searchQuery = service.Search.List(parts);
            searchQuery.ChannelId = channelId;
            searchQuery.MaxResults = 5;
            searchQuery.Order = SearchResource.ListRequest.OrderEnum.Date;
            var result = searchQuery.Execute();

            return result.Items
                .Where(x => x.Snippet.PublishedAt >= startDate).Select(x => new YoutubeVideoModel
                {
                    Id = x.Id.VideoId,
                    Name = x.Snippet.Title,
                    PublishedAt = x.Snippet.PublishedAt
                });
        }
    }
}

File: YoutubeQueuerJS/server/services/youtube-videos-service.js
var google = require('googleapis');
var googleAuth = require('./../google-auth');
var youtubeServiceProvider = require('./../youtube-service-provider');

var videosService = {};

function toVideoModel(baseVideo) {
    return {
        'id': baseVideo.id.videoId,
        'name': baseVideo.snippet.title,
        'publishedAt': new Date(Date.parse(baseVideo.snippet.publishedAt))
    };
}

videosService.getLatestVideosFromChannel = function(channelId, maxDate) {
    var youtube = youtubeServiceProvider.getYoutubeService();

    var requestParams = {
        part: 'id,snippet',
        maxResults: 50,
        order: 'date',
        channelId: channelId
    };

    maxDate = maxDate || new Date();
    maxDate.setHours(0, 0, 0, 0);

    var videosPromise = new Promise((resolve, reject) => {
        youtube.search.list(requestParams,
            (error, videos, response) => {
                if (error) {
                    console.error('Error while getting videos: ' + error);
                    reject(error);
                }

                if (videos) {
                    var data = videos.items.map(toVideoModel);
                    var filtered = data.filter(x => x.publishedAt.getTime() >= maxDate.getTime());
                    resolve(filtered);
                }
            });
    });

    return videosPromise;
}

module.exports = videosService;